webservice/export-catalogs.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var  ensureDb = require('dbjs/valid-dbjs')
+  , memoize = require('memoizee');
+
+module.exports = memoize(function (db) {
+	// ensure db
+	ensureDb(db);
+
+	var catalogs = [];
+	// all enums
+	var enums = db.StringLine.extensions.toArray().filter(function (extension) {
+		return extension.meta && extension.members;
+	}).map(function (extension) {
+		var catalogItem = { name: extension.__id__ };
+		catalogItem.options = extension.members.toArray().map(function (option) {
+			return { name: option, label: extension.meta[option].label };
+		});
+		return catalogItem;
+	});
+	[].push.apply(catalogs, enums);
+
+	// services
+	var services = db.BusinessProcess.extensions.toArray().map(function (bp) {
+		return { name: bp.__id__, label: bp.label };
+	});
+	catalogs.push({ services: services });
+
+	// registrations
+	var registrations = db.Registration.extensions.toArray().map(function (reg) {
```

Comment:
We don't (most of the time) create a classes for registrations (for example https://github.com/egovernment/eregistrations-guatemala/blob/master/model/business-process-company/registrations/business-patent.js#L7-L9) so this array is empty in GT. They are, in that case, available under `db.Registration.instances` but their `__id__` is not human friendly (`BusinessProcessMandate#/registrations/map/mandateInscription`). Using `reg.key` will help but may cause duplicates in Salvador.

Using both methods, `extensions` and `instances` will also cause duplicates as for each class we will have it's instances for each BP that uses it.

This have to be rethought…